### PR TITLE
Use collection extensions.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Simplify some operations using the Kotlin collection extension functions.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/userprofile/UserContributionsStats.kt [hunk 1]
```diff
@@ -53,73 +56,52 @@ object UserContributionsStats {
...
-                        descLang = descArr[1]
-                        break
-                    }
-                }
-            }
-            if (descLang.isEmpty()) {
+            val descLang = userContribution.comment.split(" ")
+                    .filter { "wbsetdescription" in it }
+                    .flatMap { it.split("|") }
+                    .firstOrNull()
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/userprofile/UserContributionsStats.kt:62]: We should get the second item from the splitted array. Maybe use `getOrNull(1)`?

File: app/src/main/java/org/wikipedia/userprofile/UserContributionsStats.kt [hunk 2]
```diff
@@ -53,73 +56,52 @@ object UserContributionsStats {
...
-                qLangMap[userContribution.title] = HashSet()
-            }
-            qLangMap[userContribution.title]!!.add(descLang)
+            qLangMap.getOrPut(userContribution.title, { HashSet() }).add(descLang)
         }
 
         return ServiceFactory.get(WikiSite(Service.WIKIDATA_URL)).getWikidataLabelsAndDescriptions(qLangMap.keys.joinToString("|"))
                 .subscribeOn(Schedulers.io())
-                .flatMap {
+                .flatMap { it ->
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/userprofile/UserContributionsStats.kt:72]: Can it be removed?

**Review by cooltey — CHANGES_REQUESTED** [review]
Thanks for the PR, I have learned a lot. Just have a few comments inline.

File: app/src/main/java/org/wikipedia/suggestededits/provider/EditingSuggestionsProvider.kt [hunk 3]
```diff
@@ -94,8 +92,9 @@ object EditingSuggestionsProvider {
                                         || !entity.sitelinks().containsKey(targetWiki.dbName())) {
                                     continue
                                 }
-                                articlesWithTranslatableDescriptionCache.push(Pair(PageTitle(entity.sitelinks()[targetWiki.dbName()]!!.title, targetWiki),
-                                        PageTitle(entity.sitelinks()[sourceWiki.dbName()]!!.title, sourceWiki)))
+                                val pair = PageTitle(entity.sitelinks()[targetWiki.dbName()]!!.title, targetWiki) to
```
[the remark this answers is not in the document]
Comment (Isira-Seneviratne) [reply app/src/main/java/org/wikipedia/suggestededits/provider/EditingSuggestionsProvider.kt:95]: Yeah, of course.

File: app/src/main/java/org/wikipedia/userprofile/UserContributionsStats.kt (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/wikipedia/userprofile/UserContributionsStats.kt:62]: I didn't know that, thanks!

**Pull request merged**